src/parquet/schema/schema-descriptor-test.cc

```diff
@@ -73,6 +73,37 @@ TEST_F(TestSchemaDescriptor, InitNonGroup) {
   ASSERT_THROW(descr_.Init(node), ParquetException);
 }
 
+TEST_F(TestSchemaDescriptor, DescriptorRepetitionValues) {
+  NodeVector fields;
+  NodePtr schema;
+
+  NodePtr item1 = Int32("item", Repetition::REQUIRED);
+  fields.push_back(item1);
+  NodePtr list(GroupNode::Make("list", Repetition::REPEATED,
+          {item1}, LogicalType::LIST));
+  fields.push_back(list);
+  NodePtr bag(GroupNode::Make("bag", Repetition::OPTIONAL, {list}));
+  fields.push_back(bag);
+
+  schema = GroupNode::Make("schema", Repetition::REPEATED, fields);
+
+  descr_.Init(schema);
+
+  size_t nleaves = 3;
+  // 3 leaves
+  ASSERT_EQ(nleaves, descr_.num_columns());
+
+  const ColumnDescriptor* col = descr_.Column(0);
+  EXPECT_TRUE(col->is_required());
+  col = descr_.Column(1);
+  EXPECT_TRUE(!col->is_required());
+  EXPECT_TRUE(col->is_repeated());
+  col = descr_.Column(2);
+  EXPECT_TRUE(!col->is_required());
+  EXPECT_TRUE(col->is_optional());
+  EXPECT_TRUE(col->is_repeated());
```

Comment:
It would be better to put these tests in the `scanner-test.cc` to check that the scanner logic is correct for leaf nodes in nested schemas -- I believe these tests are already overlapping with the other schema tests from the earlier patch. 

Per the discussion above I'm still unsure about how useful these methods for the potential confusion they introduce.